cf_xarray/accessor.py

```diff
@@ -2254,23 +2268,51 @@ def decode_vertical_coords(self, prefix="z"):
                     terms["depth_c"] * terms["s"]
                     + (terms["depth"] - terms["depth_c"]) * terms["C"]
                 )
+                # # expand dims so ordering is preserved
+                # terms["eta"] = terms["eta"].expand_dims(
+                #     dim={terms["s"].name: terms["s"]}, axis=1
+                # )
                 # z(n,k,j,i) = S(k,j,i) + eta(n,j,i) * (1 + S(k,j,i) / depth(j,i))
-                ds.coords[zname] = S + terms["eta"] * (1 + S / terms["depth"])
+                ztemp = terms["eta"] * (1 + S / terms["depth"]) + S
 
             elif stdname == "ocean_s_coordinate_g2":
                 # make sure all necessary terms are present in terms
                 # (depth_c * s(k) + depth(j,i) * C(k)) / (depth_c + depth(j,i))
                 S = (terms["depth_c"] * terms["s"] + terms["depth"] * terms["C"]) / (
                     terms["depth_c"] + terms["depth"]
                 )
+                # # expand dims so ordering is preserved
+                # terms["eta"] = terms["eta"].expand_dims(
+                #     dim={terms["s"].name: terms["s"]}, axis=1
+                # )
                 # z(n,k,j,i) = eta(n,j,i) + (eta(n,j,i) + depth(j,i)) * S(k,j,i)
-                ds.coords[zname] = terms["eta"] + (terms["eta"] + terms["depth"]) * S
+                ztemp = terms["eta"] + (terms["eta"] + terms["depth"]) * S
+
+            elif stdname == "ocean_sigma_coordinate":
+                # # expand dims so ordering is preserved
+                # terms["eta"] = terms["eta"].expand_dims(
+                #     dim={terms["sigma"].name: terms["sigma"]}, axis=1
+                # )
+                # z(n,k,j,i) = eta(n,j,i) + sigma(k)*(depth(j,i)+eta(n,j,i))
+                ztemp = terms["eta"] + terms["sigma"] * (terms["depth"] + terms["eta"])
 
             else:
                 raise NotImplementedError(
                     f"Coordinate function for {stdname!r} not implemented yet. Contributions welcome!"
                 )
 
+            # add Axis attribute
+            ztemp.attrs["axis"] = "Z"
+            ds.coords[zname] = ztemp
+            # reorder to time x depth x lat/Y x lon/X
+            ds.coords[zname].cf.transpose(
```

Comment:
If this is erroring it's probably because cf-xarray can't identify the T, Z, Y, X axes.

Reply:
Yeah.